contracts/verifiers/ZKPVerifier.sol

```diff
@@ -25,86 +45,69 @@ contract ZKPVerifier is IZKPVerifier, Ownable {
         uint256[2] calldata c
     ) public override returns (bool) {
         require(
-            requestValidators[requestId] != ICircuitValidator(address(0)),
+            _requests[requestId].validator != ICircuitValidator(address(0)),
             "validator is not set for this request id"
         ); // validator exists
-        require(requestQueries[requestId].queryHash != 0, "query is not set for this request id"); // query exists
 
-        _beforeProofSubmit(requestId, inputs, requestValidators[requestId]);
+        _beforeProofSubmit(requestId, inputs, _requests[requestId].validator);
 
         require(
-            requestValidators[requestId].verify(
-                inputs,
-                a,
-                b,
-                c,
-                requestQueries[requestId].queryHash
-            ),
+            _requests[requestId].validator.verify(inputs, a, b, c, _requests[requestId].data),
             "proof response is not valid"
         );
 
         proofs[msg.sender][requestId] = true; // user provided a valid proof for request
 
-        _afterProofSubmit(requestId, inputs, requestValidators[requestId]);
+        _afterProofSubmit(requestId, inputs, _requests[requestId].validator);
         return true;
     }
 
     function getZKPRequest(
         uint64 requestId
-    ) public view override returns (ICircuitValidator.CircuitQuery memory) {
-        return requestQueries[requestId];
+    ) public view override returns (IZKPVerifier.ZKPRequest memory) {
+        return _requests[requestId];
     }
 
     function setZKPRequest(
         uint64 requestId,
+        string calldata metadata,
         ICircuitValidator validator,
-        uint256 schema,
-        uint256 claimPathKey,
-        uint256 operator,
-        uint256[] calldata value
+        bytes calldata data
     ) public override onlyOwner returns (bool) {
```

Comment:
Why do we need to return bool?

Reply:
that response is verified.  I do not think that we should relay on revert and explicitly check the result